app/src/main/cpp/archive.c

```diff
@@ -488,9 +488,45 @@ Java_com_hippo_ehviewer_jni_ArchiveKt_extractToFd(JNIEnv *env, jclass thiz,
 
 JNIEXPORT void JNICALL
 Java_com_hippo_ehviewer_jni_ArchiveKt_releaseByteBuffer(JNIEnv *env, jclass thiz,
-                                                                      jobject buffer) {
+                                                        jobject buffer) {
     EH_UNUSED(thiz);
     void *addr = (*env)->GetDirectBufferAddress(env, buffer);
     size_t size = (*env)->GetDirectBufferCapacity(env, buffer);
     mempool_release_pages(addr, size);
 }
+
+JNIEXPORT jboolean JNICALL
+Java_com_hippo_ehviewer_jni_ArchiveKt_archiveFdBatch(JNIEnv *env, jclass clazz, jintArray fd_batch,
+                                                     jobjectArray names, jint arc_fd, jint size) {
+    struct archive *arc = archive_write_new();
+    struct stat64 st;
+    char buff[8192];
+    jint fdBatch[size];
+    (*env)->GetIntArrayRegion(env, fd_batch, 0, size, fdBatch);
+    archive_write_set_format_zip(arc);
+    archive_write_zip_set_compression_store(arc);
+    archive_write_open_fd(arc, arc_fd);
+    struct archive_entry *entry = archive_entry_new();
+    for (int i = 0; i < size; i++) {
+        int fd = fdBatch[i];
+        jobject name = (*env)->GetObjectArrayElement(env, names, i);
+        const char *cname = (*env)->GetStringUTFChars(env, name, false);
+        archive_entry_set_pathname(entry, cname);
+        (*env)->ReleaseStringUTFChars(env, name, cname);
+        fstat64(fd, &st);
+        archive_entry_set_size(entry, st.st_size);
+        archive_entry_set_filetype(entry, AE_IFREG);
+        archive_entry_set_perm(entry, 0644);
```

Comment:
nit: set mtime.

Reply:
I think we could use archive_entry_copy_stat(entry, &st)?

Reply:
Done

Reply:
Yes, though internal and external storage have different permissions, better to overwrite it to 644.